packages/ts-transformers/tsconfig.json

```diff
@@ -4,17 +4,22 @@
     "module": "commonjs",
     "moduleResolution": "node",
     "lib": ["es2020", "DOM"],
+    "forceConsistentCasingInFileNames": true,
+
     "strict": true,
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitOverride": true,
+    "noImplicitReturns": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "noUncheckedIndexedAccess": true,
```

Comment:
This is one of the strict options I'm unsure of. It seems like it mostly resulted in adding non-null assertions. Did you find the other cases it caught useful - were they actual bugs?

Reply:
It did actually catch some bugs! In e.g. the `customElement` decorator transform we had:

```ts
const [arg0] = decorator.expression.arguments;
if (!ts.isStringLiteral(arg0)) {
  return;
}
```

If the decorator had no arguments, `ts.isStringLiteral()` would throw, because it tries to access `.kind` on `undefined`. Turning on `noUncheckedIndexedAccess` showed me that we needed an extra guard here.

Even though type checking would also catch this particular error -- since the `customElement` decorator is typed to require an argument -- compilation would still crash during transform, since type checking doesn't block transformation.

It does require adding `!`s on a lot of safe for loops, but I feel it's worth it. I've definitely made mistakes in this area in the past too.